src/WebJobs.Script.WebHost/Security/KeyManagement/SimpleKubernetesClient.cs

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    public class SimpleKubernetesClient : IKubernetesClient
+    {
+        private const string NamespaceFile = "/run/secrets/kubernetes.io/serviceaccount/namespace";
+        private const string TokenFile = "/run/secrets/kubernetes.io/serviceaccount/token";
+        private const string CaFile = "/run/secrets/kubernetes.io/serviceaccount/ca.crt";
```

Comment:
When you give a Pod an identity in kubernetes, you get a `token` file file created there that you can use to reflect that identity. You also get a `namespace` file that contains the current namespace for the pod, and a `ca.crt` file containing the CA used to sign the api-server TLS cert.

Reply:
This will definitely be different for Windows Containers on Kubernetes, but we don't have any Windows kubernetes support yet.